provider/datagen/src/transform/cldr/displaynames/language.rs

```diff
@@ -46,22 +46,46 @@ impl IterableDataProvider<LanguageDisplayNamesV1Marker> for crate::DatagenProvid
     }
 }
 
-/// Substring used to denote alternative region names data variants for a given region. For example: "BA-alt-short", "TL-alt-variant".
-const ALT_SUBSTRING: &str = "-alt-";
+/// Substring used to denote short display names data variants for a given language. For example: "az-alt-short".
+const ALT_SHORT_SUBSTRING: &str = "-alt-short";
+/// Substring used to denote long display names data variants for a given language. For example: "az-alt-long".
+const ALT_LONG_SUBSTRING: &str = "-alt-long";
+/// Substring used to denote menu display names data variants for a given language. For example: "az-alt-menu".
+const ALT_MENU_SUBSTRING: &str = "-alt-menu";
 
 impl From<&cldr_serde::language_displaynames::Resource> for LanguageDisplayNamesV1<'static> {
     fn from(other: &cldr_serde::language_displaynames::Resource) -> Self {
         let mut names = ZeroMap::new();
+        let mut short_names = ZeroMap::new();
+        let mut long_names = ZeroMap::new();
+        let mut menu_names = ZeroMap::new();
         for lang_data_entry in other.main.0.iter() {
             for entry in lang_data_entry.1.localedisplaynames.languages.iter() {
-                let region = String::from(entry.0);
-                if !region.contains(ALT_SUBSTRING) {
+                let mut region = String::from(entry.0);
```

Comment:
`String::truncate` cuts off everything after the match, so if there are tags after `ALT_SHORT_SUBSTRING`, they will be discarded. If you expect `ALT_SHORT_SUBSTRING` to appear at the end, you don't have to allocate a `mut String`:
```rust
                if let Some(region) = entry.0.strip_suffix(ALT_SHORT_SUBSTRING) {
                    let key = UnvalidatedStr::from_str(region);
                    short_names.insert(key, entry.1.as_ref());
                } else if let Some(region) = entry.0.strip_suffix(ALT_LONG_SUBSTRING) {
                    let key = UnvalidatedStr::from_str(region);
                    long_names.insert(key, entry.1.as_ref());
                } else if let Some(region) = entry.0.strip_suffix(ALT_MENU_SUBSTRING) {
                    let key = UnvalidatedStr::from_str(region);
                    menu_names.insert(key, entry.1.as_ref());
                } else {
                    let key = UnvalidatedStr::from(&entry.0);
                    names.insert(key, entry.1.as_ref());
                }
```

Reply:
ALT_SHORT_SUBSTRING and other alt strings do appear at the end. But as truncate modifies the original string, I do need to allocate 'mut string'? Or are you suggesting to use a different method that doesn't change the original string?

Reply:
Look at the code I posted. To get a substring of a `&str`, you don't need to allocate a `String`. You can do slicing `&foo[x..y]`, or use methods like `strip_suffix` .

Reply:
That makes sense. Thanks for explaining! Added the change.